gas/generation/prompt_generator.py

```
@@ -61,29 +61,52 @@ def __init__(
 
     def load_vlm(self) -> None:
         """
-        Load the vision-language model for image annotation.
+        Load the vision-language model for image annotation with local-first loading.
         """
         bt.logging.debug(f"Loading caption generation model {self.vlm_name}")
-        self.vlm_processor = Blip2Processor.from_pretrained(
-            self.vlm_name, torch_dtype=torch.float32
-        )
-        self.vlm = Blip2ForConditionalGeneration.from_pretrained(
-            self.vlm_name, torch_dtype=torch.float32
-        )
+
+        try:
+            bt.logging.info(f"Attempting to load {self.vlm_name} from local cache...")
+            self.vlm_processor = Blip2Processor.from_pretrained(
+                self.vlm_name, torch_dtype=torch.float32, local_files_only=True
+            )
+            self.vlm = Blip2ForConditionalGeneration.from_pretrained(
+                self.vlm_name, torch_dtype=torch.float32, local_files_only=True
+            )
+        except (OSError, ValueError) as e:
+            bt.logging.info(f"Model not in local cache, downloading from HuggingFace...")
+            self.vlm_processor = Blip2Processor.from_pretrained(
+                self.vlm_name, torch_dtype=torch.float32
+            )
+            self.vlm = Blip2ForConditionalGeneration.from_pretrained(
+                self.vlm_name, torch_dtype=torch.float32
```

Potentially an issue:

Local-first loading added for BLIP2 and LLM. Issues: processors/tokenizers rarely accept torch_dtype; also local load except clauses are narrow (OSError, ValueError) and may miss TypeError/HF errors.

---

Good catch. `torch_dtype` was being set prior to this pr and wasn't causing any issues but will remove. Also would rather have this crash if the exception isn't related to the model not being present locally to surface any unexpected issues more quickly (rather than failing silently or logging stuff that we/validators don't see)